src/malli/core.cljc

```diff
@@ -954,7 +954,8 @@
      (-properties-schema [_ _])
      (-children-schema [_ _])
      (-into-schema [parent {:keys [closed] :as properties} children options]
-       (let [entry-parser (-create-entry-parser children opts options)
+       (let [pred? (:pred opts)
```

Comment:
I had this `(:pred opts map?)` exact implementation in first commit, and then changed to defaulting from 0 arity call, following the pattern for `:naked-keys` 😅 
I do see the point of NPE occurring when some passed opts but not `:pred`.

I will add the test as well. Is there a namespace (or preference) that is currently testing `m/-map-schema`? Otherwise, I will just create tests in `malli.core-test` ns.

Reply:
I have added tests for `m/-map-schema` with default pred and custom pred 🙂